app/main.py

```diff
@@ -1,6 +1,44 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Callable:
```

Comment:
The `wrapper` function is expected to return a `Callable`, but it actually returns the result of the `func`, which may not be a `Callable`. Consider updating the return type hint to match the actual return type of the `func`.